src/compas_timber/_fabrication/lap.py

```diff
@@ -0,0 +1,510 @@
+        ref_edges = [offseted_ref_edge, ref_edge]
+        # find the intersection points of the planes with the reference edges and calculate the distances from the start of the beam
+        x_distances = []
+        for plane, edge in zip(planes, ref_edges):
+            if intersection_line_plane(edge, plane) is None:
+                raise ValueError("One of the planes does not intersect with the beam.")
+
+            intersection_point = Point(*intersection_line_plane(edge, plane))
+            x_distances.append(distance_point_point(edge.start, intersection_point))
+        if orientation == OrientationType.END:
+            return max(x_distances)
+        else:
+            return min(x_distances)
+
+    @staticmethod
+    def _calculate_angle(ref_side, planes):
+        # vector rotation direction of the plane's normal in the vertical direction
+        angle_vector = Vector.cross(ref_side.normal, planes[0].normal)
+        return abs(angle_vectors_signed(ref_side.xaxis, angle_vector, ref_side.normal, deg=True))
+
+    @staticmethod
+    def _calculate_length(planes, ref_side, ref_edge, depth, angle):
+        # calculate the length of the lap based on the intersection points of the planes with the reference edge
+        # the length is the perpendicular distance between the two planes
+        vect = ref_side.yaxis.cross(planes[0].normal)
+        intersection_pts = [Point(*intersection_line_plane(ref_edge, plane)) for plane in planes]
+        dist = distance_point_point(*intersection_pts)
+        # calculate the compansation distance based on the angle of intersection
+        compensation_angle = angle_vectors_signed(vect, ref_side.xaxis, ref_side.normal)
+        dist += depth / abs(math.tan(compensation_angle))
+        # calculate the perpendicular distance between the two planes
+        length = math.sin(math.radians(angle))*dist
+        return length
+
+
+    ########################################################################
+    # Methods
+    ########################################################################
+
+    def apply(self, geometry, beam):
+        """Apply the feature to the beam geometry.
+
+        Parameters
+        ----------
+        geometry : :class:`~compas.geometry.Brep`
+            The beam geometry to be cut.
+        beam : :class:`compas_timber.elements.Beam`
+            The beam that is cut by this instance.
+
+        Raises
+        ------
+        :class:`~compas_timber.elements.FeatureApplicationError`
+            If the cutting plane does not intersect with beam geometry.
+
+        Returns
+        -------
+        :class:`~compas.geometry.Brep`
+            The resulting geometry after processing
+
+        """
+        # type: (Brep, Beam) -> Brep
+        box = self.volume_from_params_and_beam(beam)
+        lap_volume = Brep.from_box(box)
+        try:
+            return geometry - lap_volume
+        except BrepTrimmingError:
```

Comment:
don't think that's the right error here. check what `RhinoBrep.from_boolean_difference()` raises when it fails to produce anything. if I remember right it's `IndexError` (as trying to index the first result).

Reply:
I think the correct one would be `BrepError` which "Represents a generic error in the Brep context"

Reply:
check out https://github.com/compas-dev/compas/blob/46b286311ccf0212f65fa202684c978a1d6bf560/src/compas/geometry/brep/brep.py#L672

currently if no results come back from the boolean operation there'll be an `IndexError` occurring due to our implementation always trying to index the first result.

I agree this should throw some kind of a `BrepError` instead, but that'll need to be changed in core.